mount/mount_unix.go

```diff
@@ -37,35 +39,56 @@ func Unmount(target string) error {
 	}
 }
 
-// RecursiveUnmount unmounts the target and all mounts underneath, starting
-// with the deepest mount first. The argument does not have to be a mount
-// point itself.
-func RecursiveUnmount(target string) error {
-	// Fast path, works if target is a mount point that can be unmounted.
+// UnmountAll unmounts all mounts and submounts underneath parent,
+func UnmountAll(parent string) error {
+	// Get all mounts in "parent"
+	mounts, err := mountinfo.GetMounts(mountinfo.PrefixFilter(parent))
+	if err != nil {
+		return err
+	}
+
+	// Fast path: try to unmount top-level mounts first. This works if target is
+	// a mount point that can be unmounted.
 	// On Linux, mntDetach flag ensures a recursive unmount.  For other
 	// platforms, if there are submounts, we'll get EBUSY (and fall back
-	// to the slow path). NOTE we do not ignore EINVAL here as target might
-	// not be a mount point itself (but there can be mounts underneath).
-	if err := unix.Unmount(target, mntDetach); err == nil {
-		return nil
+	// to the slow path). We're not using RecursiveUnmount() here, to avoid
+	// repeatedly calling mountinfo.GetMounts()
```

Comment:
I think this comment is no longer valid (as RecursiveUnmount itself is using this function).